src/rgw/rgw_lc.cc

```diff
@@ -2642,17 +2642,19 @@ int RGWLC::LCWorker::schedule_next_start_time(utime_t &start, utime_t& now)
 {
   int secs;
 
-  if (cct->_conf->rgw_lc_debug_interval > 0) {
-	secs = start + cct->_conf->rgw_lc_debug_interval - now;
+  if (const auto interval = cct->_conf->rgw_lc_debug_interval; interval > 0) {
+	// schedule for the next interval
+	const auto remainder = now.sec() % interval;
+	secs = interval - remainder;
```

Comment:
my first attempt was closer to the existing logic:
```diff
-	secs = start + cct->_conf->rgw_lc_debug_interval - now;
+	const auto remainder = start.sec() % interval;
+	secs = start - remainder + interval - now;
```
but returned `secs = 0` for the vast majority of calls, sending `LCWorker` into a tight loop with no sleeps

it now ignores `start`, instead returning seconds until the next interval after `now`